network.tf

```diff
@@ -62,7 +62,7 @@ resource "aws_subnet" "public" {
 }
 
 resource "aws_subnet" "private" {
-  count                   = length(var.private_subnet_cidrs)
+  count                   = length(var.private_subnet_ids) > 0 ? 0 : length(var.private_subnet_cidrs)
```

Comment:
Don't want to change anything but seems like a for_each is the right thing to use here. Maybe something we can do later. LGTM